lib/imap-notifier.js

```diff
@@ -393,19 +393,29 @@ class ImapNotifier extends EventEmitter {
         }
 
         let rlkey = 'lim:' + data.service;
-        this.counters.limitedcounter(rlkey, data.user, 1, data.limit || 15, (err, res) => {
-            if (err) {
-                return callback(err);
-            }
+        const { closed: socketClosed, connecting, destroyed } = data.session.socket;
```

Comment:
Just in case `data.session` is no longer available, use something like `data.session?.socket` instead here